toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -956,28 +969,24 @@ func hydrateFromRemoteSource(fileHydrationState map[string]bool, newSourceDir st
 			<-netOpsSemaphore
 		}
 
+		// We may intentionally fail early due to a cancellation signal, stop immediately if that is the case.
 		if cancelled {
 			err = errPackerCancelReceived
 			return
 		}
 
-		if err != nil {
-			// We may intentionally fail early due to a cancellation signal, stop immediately if that is the case.
+		if internalErr != nil {
+			logger.Log.Errorf("Failed to download (%s). Error: %s", url, internalErr)
+			err = errOther
```

Comment:
I think this error just gets swallowed on the next pass through the loop. If I recall correctly the error state for each file is tracked by `fileHydrationState[fileName] = true` towards the end of the loop.

Reply:
You're right, I wanted to preserve the error but a cancel (or anything else we might do here in the future) would override it. I haven't noticed that `fileHydrationState` already does what I needed.